extensions/strategies/trendline/strategy.js

```diff
@@ -1,113 +1,66 @@
-    this.option('periodLength', 'period length', String, '30s')
-    this.option('lastpoints', 'Number of trades for short trend average', Number, 100)
-    this.option('avgpoints', 'Number of trades for long trend average', Number, 1000)
-    this.option('lastpoints2', 'Number of trades for short trend average', Number, 10)
-    this.option('avgpoints2', 'Number of trades for long trend average', Number, 100)
-    this.option('min_periods', 'Basically avgpoints + a BUNCH of more preroll periods for anything less than 5s period', Number, 15000)
-    this.option('markup_sell_pct', 'test', Number, 0)
-    this.option('markdown_buy_pct', 'test', Number, 0)
+    this.option('period', 'period length, set poll trades to 100ms, poll order 1000ms. Same as --period_length', String, '1h')
+    this.option('period_length', 'period length, set poll trades to 100ms, poll order 1000ms. Same as --period', String, '1h')
+    this.option('avgpoints', 'Trades for array 2 to be calculated stddev and mean from', Number, 100)
+    this.option('lastpoints', 'Trades for array 2 to be calculated stddev and mean from', Number, 5)
+    this.option('min_periods', 'min_periods', Number, 150)
+    this.option('order_adjust_time', 'Order Adjust Time', Number, 30000)
+    this.option('trend', 'Float number 1-2 would be increasing trend', Number, 1.1)
   },
-
   calculate: function () {
-
   },
-
   onPeriod: function (s, cb) {
-    ema(s, 'trendline', s.options.trendline)
+    ema(s, 'stddev', s.options.stddev)
     var tl1 = []
     var tls = []
-    var tll = []
-    if (s.lookback[s.options.avgpoints + 2000]) {
-      for (let i = 0; i < s.options.avgpoints + 1000; i++) { tl1.push(s.lookback[i].close) }
+    if (s.lookback[s.options.min_periods]) {
+      for (let i = 0; i < s.options.avgpoints + 10; i++) { tl1.push(s.lookback[i].close) }
       for (let i = 0; i < s.options.lastpoints; i++) { tls.push(s.lookback[i].close) }
-      for (let i = 0; i < s.options.avgpoints; i++) { tll.push(s.lookback[i].close) }
 
       var chart = tl1
 
       var growth = trend(chart, {
         lastPoints: s.options.lastpoints,
         avgPoints: s.options.avgpoints,
         avgMinimum: 0,
-        reversed: true
+        reversed: false
       })
-      var growth2 = trend(chart, {
-        lastPoints: s.options.lastpoints2,
-        avgPoints: s.options.avgpoints2,
-        avgMinimum: 0,
-        reversed: true
-      })
-
-      s.stats = growth
-      s.growth = growth > 1
-      s.stats2 = growth2
-      s.growth2 = growth2 > 1
-      s.stdevs = stats.stdev(tls)
-      s.stdevl = stats.stdev(tll)
-      s.means = math.mean(tls)
-      s.meanl = math.mean(tll)
-      s.pcts = s.stdevs / s.means
-      s.pctl = s.stdevl / s.meanl
-      s.options.markup_sell_pct = math.mean(s.pcts, s.pctl) * 100
-      s.options.markdown_buy_pct = math.mean(s.pcts, s.pctl) * 100
-      s.accel = growth > oldgrowth
-      oldgrowth = growth
+    
+      global.direc = growth > s.options.trend
```

Comment:
why you are refering here to global-object? if copying through periods you could also store it to s in general or look up the old value in s.lookback.